fix(modal-checkout): style improvements to single-gateway

miguelpeixe opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

NPPD-485

Related: https://github.com/Automattic/newspack-plugin/pull/4220

Remove the box from single-gateway checkouts:

<img width="686" height="484" alt="image" src="https://github.com/user-attachments/assets/f93dc68f-3634-4ac9-a19e-c8abc41849af" />

<img width="691" height="802" alt="image" src="https://github.com/user-attachments/assets/3129cee1-4871-4752-8ff4-193c8369c07c" />

<img width="712" height="1135" alt="image" src="https://github.com/user-attachments/assets/da2678e1-f0e8-4ff7-81fd-1c70e27e3ea9" />

### How to test the changes in this Pull Request:

1. Checkout this branch and https://github.com/Automattic/newspack-plugin/pull/4220
2. Make sure you have only Stripe as your payment gateway
3. Go through the modal checkout flow via a checkout button block
4. Confirm the card input is now borderless and the gateway title is hidden
5. Confirm you are able to fill the payment information and complete the checkout
6. Change your Woo setup to enable Woo Payments in sandbox mode so you have 2 gateways
7. Go through the checkout flow again and confirm the bordered style is preserved, you can select the different gateways and complete the checkout flow without issues
8. Deactivate the Stripe gateway and go through the checkout again
9. Confirm the Woo Payments gateway renders well in the new borderless stytle

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

miguelpeixe committed:
fix(modal-checkout): style improvements to single-gateway

thomasguillot committed:
Merge branch 'trunk' into fix/modal-checkout-card-input-style-tweaks

thomasguillot committed:
fix: font-sizes

thomasguillot committed:
Merge branch 'trunk' into fix/modal-checkout-card-input-style-tweaks

thomasguillot committed:
fix: more font-sizes and margins

thomasguillot commented:
@miguelpeixe Thank you for the PR. I made some tiny tweaks here and there, mainly regarding font-size/line-height and margins.

Couple of questions:

* Is it possible to add a "Cancel" ghost button for the change payment method modal?
* The "Back" buttons flow are strange.
    * For the Change Subscription, I'd expect to go back to the Subscription selection screen, not my Billing details.
    * For the Renew now, it doesn't make sense to be able to go back as well. Why would anyone change their billing details like this? I'd expect a Close button instead.

<img width="1364" height="1098" alt="Screenshot 2025-10-06 at 17 01 21@2x" src="https://github.com/user-attachments/assets/af4d0aca-fbb3-449c-b61d-274ef47571e2" />

https://github.com/user-attachments/assets/7cc79b4e-63f7-4f97-a8d0-b6ea1910e345

miguelpeixe committed:
Merge branch 'trunk' into fix/modal-checkout-card-input-style-tweaks

miguelpeixe committed:
feat: add "Cancel" button to change payment method

miguelpeixe commented:
> Is it possible to add a "Cancel" ghost button for the change payment method modal?

Added in e255b678b18b660a9792014a9f99ff3779bd5a68. Because the page refreshes on modal close, the most straightforward way to implement this is to link it back to the subscription page.

> The "Back" buttons flow are strange.
> For the Change Subscription, I'd expect to go back to the Subscription selection screen, not my Billing details.
> For the Renew now, it doesn't make sense to be able to go back as well. Why would anyone change their billing details like this? I'd expect a Close button instead.

That's inheriting the standard 2-step modal checkout behavior. Would it be better if we changed the button label to "Edit billing information" rather than back? Perhaps accompanied by a "Cancel" or"Close" as well?

thomasguillot commented:
> That's inheriting the standard 2-step modal checkout behavior. Would it be better if we changed the button label to "Edit billing information" rather than back? Perhaps accompanied by a "Cancel" or"Close" as well?

Yes, wfm, thank you @miguelpeixe (cancel for consistency)

miguelpeixe committed:
Merge branch 'trunk' into fix/modal-checkout-card-input-style-tweaks

miguelpeixe committed:
feat: introduce Cancel button and relabel "Back" to "Edit billing information"

miguelpeixe commented:
Implemented in 5010817d396b519e8486c9970e8b7c2ddb18d80a. I changed the Edit billing information button to secondary instead of ghost. Let me know if both Edit billing and Cancel should be ghost:

<img width="659" height="839" alt="image" src="https://github.com/user-attachments/assets/72d61a0c-27e1-43b2-a527-346bc9bf4488" />

miguelpeixe committed:
chore: remove unused label

thomasguillot commented:
This is perfect @miguelpeixe

thomasguillot commented:
2 more small issues I noticed:

<img width="1478" height="1064" alt="Screenshot 2025-10-07 at 20 06 03@2x" src="https://github.com/user-attachments/assets/1217401d-7a00-413c-861a-cdb435799acd" />

This is if I edit my billing details, the "Cancel" button shouldn't have a margin-bottom.

<img width="1508" height="1124" alt="Screenshot 2025-10-07 at 20 06 30@2x" src="https://github.com/user-attachments/assets/067f5fa7-e611-492d-a81f-d1ae2c558cdc" />

And there's a similar issue here.

miguelpeixe committed:
fix: cancel button margin

miguelpeixe commented:
Good catch! Fixed in eddf6deaa8a5f759f020cc188f4ee1275806e76e

thomasguillot commented:
Thank you @miguelpeixe!

miguelpeixe committed:
Merge branch 'trunk' into fix/modal-checkout-card-input-style-tweaks

dkoo committed:
Merge branch 'trunk' into fix/modal-checkout-card-input-style-tweaks
